lms/djangoapps/course_api/views.py

```diff
@@ -211,6 +211,28 @@ class CourseListView(DeveloperErrorViewMixin, ListAPIView):
     pagination_class = NamespacedPageNumberPagination
     serializer_class = CourseSerializer
 
+    def __init__(self):
+        super(CourseListView, self).__init__()
+        self.courses_list = list()
+
+    def get_serializer_context(self):
+        context = super(CourseListView, self).get_serializer_context()
+
+        calculate_completion = self.request.GET.get('calculate_completion') in ['true', 'True']
+        if calculate_completion:
+            context['courses_completion'] = self.get_courses_completion()
+
+        context['calculate_completion'] = calculate_completion
+
+        return context
+
+    def get_courses_completion(self):
+        courses_completion = dict()
+        for course in self.courses_list:
+            course_id_str = str(course.id)
+            courses_completion[course_id_str] = bool(is_student_pass(self.request.user, self.request, course_id_str))
+        return courses_completion
```

Comment:
Why do you do the calculation here. Its quite counterintuitive. What I suggest is move this logic to the serializer and then calculate the completion for each course individually.